include/dxc/dxcisense.h

```diff
@@ -728,6 +728,10 @@ IDxcSourceLocation : public IUnknown
     _Out_opt_ unsigned* pLine,
     _Out_opt_ unsigned* pCol,
     _Out_opt_ unsigned* pOffset) = 0;
+  virtual HRESULT STDMETHODCALLTYPE GetPresumedLocation(
```

Comment:
You'd want to at least put this after `IsNull` to keep the vtable stable.

Also, it seems `tools/clang/tools/dotnetc/DotNetDxc.cs` has fallen behind, FWIW.

Reply:
I've moved `GetPresumedLocation` to the end of `IDxcSourceLocation`.

`DotNetDxc` is an interesting one, because I don't think I'm actually going to be able to use it myself. I'm building my LSP stuff in .NET Core, and .NET Core doesn't support COM interop on non-Windows platforms. So COM interop needs to be done manually (i.e. `Marshal.GetDelegateForFunctionPointer`). I'm planning to use [SharpGenTools](https://github.com/SharpGenTools/SharpGenTools) to automatically generate these bindings from the C++ header `dxcisense.h`. SharpGenTools is particularly nice because it makes it easy to use the more efficient `calli` instead of `Marshal.GetDelegateForFunctionPointer`, which is arguably the best way to do COM interop even on Windows.

That said, I don't mind updating `DotNetDxc` with my changes if you'd like?

Reply:
Just for completeness, the ideal scenario from my point of view is that the step I mentioned above (generating .NET Core-compatible C# bindings from C++ headers, and perhaps even creating a NuGet package with these bindings) could be a part of the official dxc build. That would make it much easier to keep native and C# versions in sync and updated.

But I'm not expecting an immediate decision on that, and this issue probably isn't the right place to have that discussion anyway :)

Reply:
@tgjones In the full diff, it looks like you've added non-trailing virtual functions and moved IsNull. Can you double-check? I'll merge once that's good.

Reply:
Oops, I moved the wrong function to the end of `IDxcSourceLocation`. Fixed now.